packages/office-addin-dev-settings/src/dev-settings-windows.ts

```diff
@@ -178,7 +178,7 @@ export async function registerAddIn(manifestPath: string, registration?: string)
   const appsInManifest = getOfficeAppsForManifestHosts(manifest.hosts);
 
   // Register using the service
-  if (appsInManifest.indexOf(OfficeApp.Outlook) >= 0) {
+  if (manifestPath.endsWith(".json") || appsInManifest.indexOf(OfficeApp.Outlook) >= 0) {
```

Comment:
If you have the manifest info which contains the manifest type, it might be better to use that rather than just checking the extension. 

What happens for Mac with JSON manifests? Is this only being implemented on Windows now, or does Mac not also need changes?

Reply:
It might be better to add an optional parameter for this function about how the registration should be done, e.g. `registerUsingService: boolean` and the caller can have the logic about whether to use the service based on manifest type or whether Outlook or other conditions. The code here could error if Outlook and registerUsingService is false. For undefined, it would treat the value as false. But later the default could change to true unless explicitly false.

Reply:
Yes, I have change it to use the ManifestInfo object(manifestType). Thanks for your heads up!